### ISSUE TITLE
User search just found results with display name

### ISSUE THREAD

**Issue by GustavoContreiras** [mdanics/fluttergram#37]
Searched for a user called "Pendulum Pala" with username "pendulumpala".
Typing:
Pendulum -> found
pendulum -> not found
pendulumpala -> not found
pendulum pala -> not found
pendulum Pala -> not found
Pendulum Pala -> found
Pala -> found
pala -> not found

I'm gonna try to improve that search and then I come back here.

**Comment by mdanics** [comment]
Yeah the searching is not ideal. In order to do full text search we would need to implement [Algolia search](https://firebase.google.com/docs/firestore/solutions/search) which when I looked into it was on a separate pricing plan than firebase, and I wanted to keep project running for free. 

If there's another alternative though I'd be interested in exploring it.

**Comment by GustavoContreiras** [comment]
Isnt it possible to get the text input and make variations of it? Something like, if I type: gustavo contreiras
Create new strings with variations:
Gustavo, GUSTAVO, gustavo, Contreiras, CONTREIRAS, Gustavo Contreiras, gustavocontreiras, ....
Then make one search for each generated word and return everything
